pkg/encryption/cipher.go

```diff
@@ -17,6 +17,29 @@ import (
 	"time"
 )
 
+// SecretBytes attempts to base64 decode the secret, if that fails it treats the secret as binary
+func SecretBytes(secret string) []byte {
+	b, err := base64.URLEncoding.DecodeString(addPadding(secret))
+	if err == nil {
+		return []byte(addPadding(string(b)))
+	}
+	return []byte(secret)
+}
+
+func addPadding(secret string) string {
```

Comment:
Can you add a quick comment to this?

Reply:
Truthfully, I'm a little confused about the motivation of this function and how its used above -- I just left it to preserve existing functionality in the code shift from the old two areas to here.

Is it just for handling possibly base64 that is unpadded? Then why the case adding `===` (doesn't happen in B64, single trailing byte case in original data being 8 bits expands to 2 chars of 6 bits each in the encoding).

But then why is it reapplied to the decoded bytes before they are used as a secret? Is it more appropriate to check for 16,24,32 lengths and error? This seems like a hidden way for users to unintentionally get 13 byte secrets (which pad to 16), which just feels low to then use in encryption.

Reply:
> Is it just for handling possibly base64 that is unpadded?

Yes.

> Then why the case adding === (doesn't happen in B64

Indeed. This is historical code, existing without modifications in `options.go` since mid 2016 ... trying to just get it to work if at all possible and then move on.

Reply:
> But then why is it reapplied to the decoded bytes before they are used as a secret? Is it more appropriate to check for 16,24,32 lengths and error? This seems like a hidden way for users to unintentionally get 13 byte secrets (which pad to 16), which just feels low to then use in encryption.

If this is true, then I'd consider this a bug. As I understood it, it would be used to pad any Base64 secret that wasn't a multiple of 4 in length. I don't think it should affect secrets if they are not being base64 decoded

Reply:
that part is even hackier than I remember ... it doesn't "pad with =" anything which is not base64, but anything which is base64 (with or without padding originally), it:

  1. base64-decodes to bytes
  2. converts to string
  3. adds `=` padding to the byte-string
  4. converts to bytes

So if you base64-encoded 14 random bytes, it would end up adding ascii `==` at the end and accepting it. That's not right. That second string()/addPadding() in `secretBytes()` should be removed.